perspective.el

```diff
@@ -1069,32 +1077,47 @@ PERSP-SET-IDO-BUFFERS)."
         (ibuffer))
     (ibuffer)))
 
-;; Buffer switching integration: Ivy / Counsel.
-;;;###autoload
-(defun persp-counsel-switch-buffer (arg)
-  "Like COUNSEL-SWITCH-BUFFER, but Perspective-aware.
-With a prefix arg, show buffers in all perspectives."
-  (interactive "P")
-  (unless (and (featurep 'ivy) (featurep 'counsel))
-    (error "Ivy or Counsel not loaded"))
+(defun persp--switch-buffer-1 (arg ivy-params fallback)
```

Comment:
Let's call this function `persp--switch-buffer-ivy-counsel-helper`, since it's so specific to these packages.